### PR TITLE
done

### PR DESCRIPTION
**Pull request by msselizabeth**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
adding
## What did you learn from the changes you have made?
how to build a classifier

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
all of that :)
## How were these changes tested?
VS Code
## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by fyellow — CHANGES_REQUESTED** [review]
Incomplete: Please make the following changes
Split data before standardizing
Fit scaler only on training data
Use train_test_split() on both X and y
Only grid search and train on training set
Only evaluate on test set
Use stratify=y and random_state=123 to ensure balanced and reproducible splits

**Comment by msselizabeth** [comment]
Hi, I did my best to implement your comment as I tried to follow the learning session. Please let me know if I did wrong and where I could find an example to understand. Thank you !

**Comment by fyellow** [comment]
Hello! Good job you are almost there, you are currently using train accuracy instead of test. Evaluation of model happens on the test dataset. 
Example: knn = KNeighborsClassifier(n_neighbors=21)
knn.fit(X_train_scaled, y_train)
y_pred = knn.predict(X_test_scaled)
acc = accuracy_score(y_test, y_pred)
print(f"Test Accuracy: {acc:.4f}")

Number of predictors is 13.

**Comment by msselizabeth** [comment]
Hope, this is fine :)
